frame/support/src/traits/hooks.rs

```diff
@@ -390,4 +421,65 @@ mod tests {
 			ON_IDLE_INVOCATION_ORDER.clear();
 		}
 	}
+
+	#[cfg(feature = "try-runtime")]
+	#[test]
+	fn on_runtime_upgrade_tuple() {
```

Comment:
interesting. So when you call `pre_upgrade` for a tuple of `n` elements, you should still get a `Vec<u8>` back, but this `Vec<u8>` is the encoding of `vec![x1, x2, x3]` where `x1` is the `Vec<u8>` returned by the first tuple etc. Then, I think we can be able to pass the right data back to the corresponding `post_upgrade`.

Reply:
Right, this is how it works.